rust/vectordb/src/io/object_store.rs

```diff
@@ -0,0 +1,380 @@
+use std::{
+    fmt::Formatter,
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+use bytes::Bytes;
+use futures::{stream::BoxStream, FutureExt, StreamExt};
+use lance::io::object_store::WrappingObjectStore;
+use object_store::{
+    path::Path, GetOptions, GetResult, ListResult, MultipartId, ObjectMeta, ObjectStore, Result,
+};
+
+use async_trait::async_trait;
+use tokio::{
+    io::{AsyncWrite, AsyncWriteExt},
+    task::JoinHandle,
+};
+
+#[derive(Debug)]
+struct MirrowingObjectStore {
```

Comment:
Should we call this as pass through ObjectStore?

Reply:
Btw, primary / secondary which one is fast / caching, which one is slow / persist storage?

Reply:
primary is the slow one.

I added some comments here
https://github.com/lancedb/lancedb/pull/537/files#diff-801c5c16cbc0082436e711a6275900a60849b2b01e2c8589c9b579579f81abedR48-R55